changelog/13192.feature.2.rst

```diff
@@ -0,0 +1 @@
+The context-manager form of :func:`pytest.raises` now has a ``check`` parameter that takes a callable that should return `True` when passed the raised exception if it should be accepted.
```

Comment:
I think we really need to spell out this api, for people who aren't really familiar with Pytest.  Something like 

> You can now pass :func:`with pytest.raises(check=fn): <pytest.raises>`, where ``fn`` is a function which takes a raised exception and returns a boolean.  The test fails if no exception was raised (as usual with :func:`pytest.raises`), passes if an exception is raised and ``fn`` returns True, and re-raises the exception if ``fn`` returns False (which also fails the test).

Reply:
I'm not sure the changelog needs to cater super hard to people not familiar with pytest at all, but agree it should be less technical. Though we no longer just reraise on fail, and made it clearer that it can be used in conjunction with the other parameters.